core/src/common.cpp

```diff
@@ -207,6 +218,143 @@ namespace irods::http
 		return parse_url(fmt::format("http://ignored{}", _req.target()));
 	} // parse_url
 
+	auto url_encode_body(const body_arguments& _args) -> std::string
+	{
+		auto encode_pair{[](const body_arguments::value_type& i) {
+			return fmt::format("{}={}", encode(i.first), encode(i.second));
+		}};
+
+		return std::transform_reduce(
+			std::next(std::cbegin(_args)),
+			std::cend(_args),
+			encode_pair(*std::cbegin(_args)),
+			[](const auto& a, const auto& b) { return fmt::format("{}&{}", a, b); },
+			encode_pair);
+	}
+
+	auto safe_base64_encode(std::string_view _view) -> std::string {
+		constexpr auto char_per_byte_set{4};
+		constexpr auto byte_set_size{3};
+
+		const auto max_size{char_per_byte_set * ((_view.size() + 2) / byte_set_size)};
+		auto max_size_plus_null_term{max_size + 1};
+
+		std::string encoded_data;
+		encoded_data.resize(max_size);
+
+		auto res{irods::base64_encode(reinterpret_cast<const unsigned char*>(_view.data()),
+									  _view.size(),
+									  reinterpret_cast<unsigned char*>(encoded_data.data()),
+									  &max_size_plus_null_term)};
+		if (res) {
+			log::error("{}: Failed to encode the string [{}], output may be unusable.", __func__, _view);
+		}
+		return encoded_data;
+	}
+
+	auto create_host_field(boost::urls::url_view _url, std::string_view _port) -> std::string {
+		if ((_port == "443" && _url.scheme_id() == boost::urls::scheme::https) ||
+			(_port == "80" && _url.scheme_id() == boost::urls::scheme::http)) {
+			return _url.host();
+		}
+		return fmt::format("{}:{}", _url.host(), _port);
+	}
+
+	auto create_oidc_request(boost::urls::url_view _url) -> beast::http::request<beast::http::string_body> {
+		constexpr auto http_version_number{11};
+		beast::http::request<beast::http::string_body> req{beast::http::verb::post, _url.path(), http_version_number};
+
+		const auto port{get_port_from_url(_url)};
+
+		req.set(beast::http::field::host, create_host_field(_url, *port));
+		req.set(beast::http::field::user_agent, irods::http::version::server_name);
+		req.set(beast::http::field::content_type,
+		        "application/x-www-form-urlencoded");
+		req.set(beast::http::field::accept, "application/json");
+
+		if (const auto& secret_key{irods::http::globals::oidc_configuration().find("client_secret")}; secret_key != std::end(irods::http::globals::oidc_configuration())) {
```

Comment:
Drop `&`?

`secret_key` is an iterator.

Reply:
Done.